app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,42 @@
+class PostsController < ApplicationController
+  def index
```

Comment:
interesting you want authentication restrictions on just the index route. Almost seems like CUD functionality should be protected and allow others to view whatever they want. But it totally depends on the domain you want for your scribble application.